content/blog/tyranny-of-nits/index.md

```diff
@@ -0,0 +1,208 @@
+Get someone properly [nerd sniped](https://xkcd.com/356/) and they'll spend entirely disproportionate effort trying to fix or improve open source projects.
+Because fundamentally, it's *fun*!
+
+But when it feels like they're being dragged down by pointless minutia, or can't make progress, they'll wander off in frustration and find something better to do.
+Even when you're talking about contributions done as part of their job: waste their time and next time they won't file or fix that little bug.
+Do it enough and they'll bite the bullet and migrate away.
+
+**Progress is key.** To the greatest extent you possibly can, as a maintainer and reviewer, you need to keep the ball rolling on work. Otherwise, it will stop!
+
+So far, none of this is very controversial.
+If you've been a software developer for long at all, you've been on the receiving end of nits and scope creep,
+or watched it block a feature that you've desperately awaited as a functioning PR just quietly dies.
+It's really tempting, and really easy, to just blame the reviewer here.
+If they just chilled, just let the nits slide, just fixed them themselves, didn't scope creep: then everything would be okay!
+
+But that's not a real solution for anyone other than "the nitpicking reviewer".
+Around here, **we believe in systematic solutions to systematic problems.**
+And if reading this post has left you with as many bitter memories of [great ideas that may never be](https://soasis.org/posts/a-mirror-for-rust-a-plan-for-generic-compile-time-introspection-in-rust/) as writing it left in me,
+you can be assured this is a systematic problem.
+
+## Automation for humans
+
+Axiomatically, there are two ways to solve nitpicking in code review.
+First, reduce the amount of nits that need picking.
+Ensure that code that comes in is free of obvious trivial mistakes.
+Linters and formatters are great for that!
+
+But just as importantly: invest in paying off your tech debt and clarifying your architecture.
+If it's *obvious* how to do something, you won't have to waste everyone's time explaining that someone did something the wrong way.
+Get those eager new contributors to help, seriously.
+
+Secondly, you can reduce the maount of nits that are picked.
+I don't actually think that telling reviewers "don't worry about this stuff" is generally a useful strategy.
+Instead, get them to focus on splitting apart their "would be nice" feedback from "this must be fixed" feedback.
+
+If you want to implement *any* of these changes effectively you need automation: whether that's CI for the machines or written guidelines for humans.
+[I gave a whole talk about this](https://www.youtube.com/watch?v=u3PJaiSpbmc): go watch it!
+
+## A taxonomy of code concerns
+
+Nits are in the eye of the beholder (we would make a joke here, but then [WotC](https://en.wikipedia.org/wiki/Wizards_of_the_Coast) would sue us for [trademark infringement](https://en.wikipedia.org/wiki/Beholder_(Dungeons_%26_Dragons))).
```

Comment:
The beholder thing is funny, but it's too deep a cut for most people I think, and it kinda distracts from the main post. I'd cut.